test/run-drun/pass-self.mo

```diff
@@ -0,0 +1,13 @@
+import { debugPrint; principalOfActor } = "mo:⛔";
+
+actor Self {
+    public func method() : async () { debugPrint "YESS!"; ignore principalOfActor Self };
+    func caller(_callee : shared () -> async ()) { };
+
+    debugPrint "Before!";
+    ignore principalOfActor Self;
+    caller(Self.method);
+    caller(method);
+    debugPrint (debug_show(principalOfActor Self));
+    debugPrint "So far so good!";
```

Comment:
Strange, this one doesn't generate a a type error about trailing 
debugPrint, but pass-class-self does?

Reply:
Yeah, very bizarre.